openmdao/core/problem.py

```diff
@@ -1690,7 +1699,22 @@ def check_partial_derivatives(self, out_stream=sys.stdout):
 
         # Check derivative calculations for all comps at every level of the
         # system hierarchy.
-        for comp in root.components(recurse=True):
+        allcomps = root.components(recurse=True)
+        if comps is None:
+            comps = allcomps
+        else:
+            allcompnames = set([c.pathname for c in allcomps])
+            requested = set(comps)
+            diff = requested.difference(allcompnames)
+
+            if diff:
+                msg = "The following are not valid comp names: "
```

Comment:
I thought turning a set into a list preserved the sort order, since sets are sorted.